src/main/java/com/minecolonies/structures/client/TemplateTessellator.java

```diff
@@ -0,0 +1,156 @@
+        }
+
+        preTemplateBufferBinding(rotation, mirror, drawingOffset, inTemplateOffset);
+
+        this.buffer.bindBuffer();
+
+        preTemplateDraw();
+
+        this.buffer.drawArrays(GL_QUADS);
+
+        postTemplateDraw();
+
+        this.buffer.unbindBuffer();
+
+        postTemplateBufferUnbinding();
+    }
+
+    private static void preTemplateBufferBinding(final Rotation rotation, final Mirror mirror, final Vector3d drawingOffset, final BlockPos inTemplateOffset)
+    {
+        final ITextureObject textureObject = Minecraft.getMinecraft().getTextureMapBlocks();
+        GlStateManager.bindTexture(textureObject.getGlTextureId());
+
+        GlStateManager.pushMatrix();
+        GlStateManager.translate(drawingOffset.x, drawingOffset.y, drawingOffset.z);
+
+        BlockPos rotateInTemplateOffset = inTemplateOffset.rotate(rotation);
+        GlStateManager.translate(-rotateInTemplateOffset.getX(), -rotateInTemplateOffset.getY(), -rotateInTemplateOffset.getZ());
+
+        RenderUtil.applyRotationToYAxis(rotation);
+        RenderUtil.applyMirror(mirror, inTemplateOffset);
+
+        GlStateManager.scale(HALF_PERCENT_SHRINK, HALF_PERCENT_SHRINK, HALF_PERCENT_SHRINK);
+        GlStateManager.enableBlend();
+        GlStateManager.blendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+        GlStateManager.resetColor();
+        GlStateManager.pushMatrix();
+    }
+
+    private static void preTemplateDraw()
+    {
+        GlStateManager.glEnableClientState(GL_VERTEX_ARRAY);
+        OpenGlHelper.setClientActiveTexture(OpenGlHelper.defaultTexUnit);
+        GlStateManager.glEnableClientState(GL_TEXTURE_COORD_ARRAY);
+        OpenGlHelper.setClientActiveTexture(OpenGlHelper.lightmapTexUnit);
+        GlStateManager.glEnableClientState(GL_TEXTURE_COORD_ARRAY);
+        OpenGlHelper.setClientActiveTexture(OpenGlHelper.defaultTexUnit);
+        GlStateManager.glEnableClientState(GL_COLOR_ARRAY);
+
+        GlStateManager.glVertexPointer(VERTEX_COMPONENT_SIZE, GL_FLOAT, VERTEX_SIZE, VERTEX_COMPONENT_OFFSET);
+        GlStateManager.glColorPointer(COLOR_COMPONENT_SIZE, GL_UNSIGNED_BYTE, VERTEX_SIZE, COLOR_COMPONENT_OFFSET);
+        GlStateManager.glTexCoordPointer(TEX_COORD_COMPONENT_SIZE, GL_FLOAT, VERTEX_SIZE, TEX_COORD_COMPONENT_OFFSET);
+        OpenGlHelper.setClientActiveTexture(OpenGlHelper.lightmapTexUnit);
+        GlStateManager.glTexCoordPointer(LIGHT_TEX_COORD_COMPONENT_SIZE, GL_SHORT, VERTEX_SIZE, LIGHT_TEXT_COORD_COMPONENT_OFFSET);
+        OpenGlHelper.setClientActiveTexture(OpenGlHelper.defaultTexUnit);
+
+        GlStateManager.disableCull();
+    }
+
+    private void postTemplateDraw()
+    {
+        GlStateManager.enableCull();
+
+        for (VertexFormatElement vertexformatelement : DefaultVertexFormats.BLOCK.getElements())
+        {
+            VertexFormatElement.EnumUsage vfeUsage = vertexformatelement.getUsage();
+            int k1 = vertexformatelement.getIndex();
+
+            switch (vfeUsage)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a default case to this switch. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)